src/emulator/pubsubEmulator.ts

```diff
@@ -105,20 +120,71 @@ export class PubsubEmulator implements EmulatorInstance {
       this.onMessage(topicName, message);
     });
 
-    topicTriggers.add(trigger);
-    this.triggers.set(topicName, topicTriggers);
-    this.subscriptions.set(topicName, sub);
+    triggers.push({ triggerKey, signatureType });
+    this.triggersForTopic.set(topicName, triggers);
+    this.subscriptionForTopic.set(topicName, sub);
+  }
+
+  private getRequestOptions(
+    topic: string,
+    message: Message,
+    signatureType: SignatureType
+  ): Record<string, unknown> {
+    const baseOpts = {
+      origin: `http://${EmulatorRegistry.getInfoHostString(
+        EmulatorRegistry.get(Emulators.FUNCTIONS)!.getInfo()
+      )}`,
+    };
+    if (signatureType === "event") {
+      return {
+        ...baseOpts,
+        data: {
+          context: {
+            eventId: uuid.v4(),
+            resource: {
+              service: "pubsub.googleapis.com",
+              name: `projects/${this.args.projectId}/topics/${topic}`,
+            },
+            eventType: "google.pubsub.topic.publish",
+            timestamp: message.publishTime.toISOString(),
+          },
+          data: {
+            data: message.data,
+            attributes: message.attributes,
+          },
+        },
+      };
+    } else if (signatureType === "cloudevent") {
+      const data: MessagePublishedData = {
+        message: {
+          messageId: message.id,
+          publishTime: message.publishTime,
+          attributes: message.attributes,
+          orderingKey: message.orderingKey,
+          data: message.data.toString("base64"),
+        },
+        subscription: this.subscriptionForTopic.get(topic)!.name,
+      };
+      const ce = HTTP.structured(
+        new CloudEvent({
+          type: "google.cloud.pubsub.topic.v1.messagePublished",
+          source: `//pubsub.googleapis.com/projects/${this.args.projectId}/topics/${topic}`,
```

Comment:
Really? It's all `source` and none of it is `subject`?

Reply:
🤷  That's what I saw:

![Screen Shot 2021-09-22 at 11 35 52 PM](https://user-images.githubusercontent.com/14897098/134463886-a4ef9c15-904d-4d94-83c9-baf999f4c710.png)